books/2015/pro-android-web-game-apps-2012/summary/Chapter2.md

```diff
@@ -103,13 +103,91 @@ draw an outline (`drawRect`) or a filled shape (`fillRect`). They both take
 four arguments: the top-left corner of the rectangle, its width, and its
 height.
 
-> See [listing-2-3] and [listing-2-4] for more examples.
+> See [listing 2.3] and [listing 2.4] for more examples.
 
-[listing-2-3]: http://127.0.0.1:8080/ch2/listing-2-3.html
-[listing-2-4]: http://127.0.0.1:8080/ch2/listing-2-4.html
+[listing 2.3]: http://127.0.0.1:8080/ch2/listing-2-3.html
+[listing 2.4]: http://127.0.0.1:8080/ch2/listing-2-4.html
 
 ### Paths
 
+For non-rectangular shapes, the process has one more step: one must first
+define a _path_, and then either fill it or stroke it (or both).
+
+There are four ways to define paths: lines, arcs, quadratic curves, or Bézier
+curves.
+
+To illustrate the overall process, here is the code to draw a simple line:
+
+```javascript
+ctx.beginPath();
+ctx.moveTo(50, 50); // moves the cursor; not defining anything yet
+ctx.lineTo(120, 100); // begins tracing a path; we could have more steps here
+ctx.strokeStyle = "#000"; // set color; still not displaying anything
+ctx.stroke(); // this makes the black line from 50, 50 to 120, 100 appear
+```
+
+We use this to draw a grid in [listing 2.5]:
+
+```javascript
+      var cellSize = 40;
+      ctx.beginPath();
+
+      for (var i = 8; i < 8; i++) {
```

Comment:
```suggestion
      for (var i = 0; i < 8; i++) {
```

Reply:
🤦

I corrected it in the listing, but forgot I'd also put a copy in the text. Thanks for catching it!